src/mem/mem_pool.c

```diff
@@ -0,0 +1,251 @@
+/**
+ * @file mem_pool.c  Pre-Allocated Memory pool management
+ *
+ * Copyright (C) 2025 Sebastian Reimers
+ */
+
+#include <string.h>
+
+#include <re_types.h>
+#include <re_mem.h>
+#include <re_thread.h>
+#include <re_atomic.h>
+
+
+#define DEBUG_MODULE "mem_pool"
+#define DEBUG_LEVEL 5
+#include <re_dbg.h>
+
+
+struct mem_pool {
+	RE_ATOMIC size_t next_free;
+	size_t nmemb;
+	size_t membsize;
+	struct mem_pool_entry *objs;
+	mem_destroy_h *membdh;
+};
+
+struct mem_pool_entry {
+	RE_ATOMIC bool used;
+	void *member;
+};
+
+
+static void mem_pool_destroy(void *data)
+{
+	struct mem_pool *p = data;
+
+	for (size_t i = 0; i < p->nmemb; i++) {
+		mem_deref(p->objs[i].member);
+	}
+
+	mem_deref(p->objs);
+}
+
+
+/**
+ * @brief Allocate a memory pool.
+ *
+ * This function initializes a memory pool with a specified number of elements,
+ * each of a given size. Optionally, a destructor callback can be provided
+ * to handle cleanup when a member is released or pool is destroyed.
+ *
+ * @param  poolp    Pointer to the memory pool pointer to be initialized.
+ * @param  nmemb    Number of elements to allocate in the pool.
+ * @param  membsize Size of each element in the pool.
+ * @param  dh       Optional destructor callback for pool cleanup (can be
+ *                  NULL).
```

Comment:
Maybe it is not very important. But I always thought that we do not want the dots at the end in Doxygen